From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Matt Borgerson <contact@mborgerson.com>
Date: Tue, 11 Jun 2024 16:52:14 -0700
Subject: [PATCH] 6631: Encode SlaSpace wordsize as an unsigned integer

This patch corrects a mismatch between the encoder and decoder in
wordsize signedness. AddrSpace::getWordSize() returns an unsigned
integer, so use unsigned here for consistency.
---
 Ghidra/Features/Decompiler/src/decompile/cpp/sleighbase.cc | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Ghidra/Features/Decompiler/src/decompile/cpp/sleighbase.cc b/Ghidra/Features/Decompiler/src/decompile/cpp/sleighbase.cc
index 3e0cf24dd5..85ce558782 100644
--- a/Ghidra/Features/Decompiler/src/decompile/cpp/sleighbase.cc
+++ b/Ghidra/Features/Decompiler/src/decompile/cpp/sleighbase.cc
@@ -199,7 +199,7 @@ void SleighBase::encodeSlaSpace(Encoder &encoder,AddrSpace *spc) const
 //    encoder.writeSignedInteger(sla::ATTRIB_DEADCODEDELAY, spc->getDeadcodeDelay());
   encoder.writeSignedInteger(sla::ATTRIB_SIZE, spc->getAddrSize());
   if (spc->getWordSize() > 1)
-    encoder.writeSignedInteger(sla::ATTRIB_WORDSIZE, spc->getWordSize());
+    encoder.writeUnsignedInteger(sla::ATTRIB_WORDSIZE, spc->getWordSize());
   encoder.writeBool(sla::ATTRIB_PHYSICAL, spc->hasPhysical());
   if (spc->getType() == IPTR_INTERNAL)
     encoder.closeElement(sla::ELEM_SPACE_UNIQUE);
-- 
2.45.1

